catchpoint-mixin/panels.libsonnet

```diff
@@ -0,0 +1,418 @@
+        )
+        + g.panel.timeSeries.fieldConfig.defaults.custom.withSpanNulls('true')
+        + g.panel.timeSeries.standardOptions.withUnit('ms'),
+
+      topAvgContentLoadingTimeNodeName:
+        commonlib.panels.generic.timeSeries.base.new(
+          'Top average content loading time by nodes',
+          targets=[t.topAvgContentLoadingTimeNodeName],
+          description='The top average content loading time among all node names over the specified interval.'
+        )
+        + g.panel.timeSeries.fieldConfig.defaults.custom.withSpanNulls('true')
+        + g.panel.timeSeries.standardOptions.withUnit('ms'),
+
+      topAvgRedirectsTestName:
+        commonlib.panels.generic.timeSeries.base.new(
+          'Top average redirects by tests',
+          targets=[t.topAvgRedirectsTestName],
+          description='The top average number of redirects among all test names over the specified interval.'
+        )
+        + g.panel.timeSeries.fieldConfig.defaults.custom.withSpanNulls('true'),
+
+      topAvgRedirectsNodeName:
+        commonlib.panels.generic.timeSeries.base.new(
+          'Top average redirects by nodes',
+          targets=[t.topAvgRedirectsNodeName],
+          description='The top average number of redirects among all node names over the specified interval.'
+        )
+        + g.panel.timeSeries.fieldConfig.defaults.custom.withSpanNulls('true'),
+
+      alertsPanel:
+        alertList.new('Catchpoint alerts')
+        + alertList.options.UnifiedAlertListOptions.withAlertInstanceLabelFilter(this.grafana.variables.queriesGroupSelectorAdvanced),
+
+      topErrorsByTestName:
+        commonlib.panels.generic.timeSeries.base.new(
+          'Top errors by tests',
+          targets=[t.topErrorsByTestName],
+          description='The top number of errors encountered among all test names over the specified interval.'
+        )
+        + g.panel.timeSeries.fieldConfig.defaults.custom.withSpanNulls('true'),
+
+      // Web Performance by Tests Dashboard Panels
+      pageCompletionTime:
+        commonlib.panels.generic.timeSeries.base.new(
+          'Page completion time',
+          targets=[t.pageCompletionTime, t.pageTotalLoadTime],
+          description='Time taken for the browser to fully render the page after all resources are downloaded.'
+        )
+        + g.panel.timeSeries.standardOptions.withUnit('ms')
+        + g.panel.timeSeries.fieldConfig.defaults.custom.withSpanNulls('true'),
+
+      DNSResolution:
+        commonlib.panels.generic.timeSeries.base.new(
+          'Connection and DNS resolution',
+          targets=[t.DNSResolution, t.SSLTime, t.connectTime],
+          description='Time taken to establish an SSL handshake, DNS resolution, and connect.'
+        )
+        + g.panel.timeSeries.standardOptions.withUnit('ms')
+        + g.panel.timeSeries.fieldConfig.defaults.custom.withSpanNulls('true'),
```

Comment:
this panel needs some work on the legend, as it's crowded and hard to read.

Possibly a case for full-width + right aligned table legend with values, as it's currently cropping the legend just with test data
![image](https://github.com/grafana/jsonnet-libs/assets/1282515/f53c376e-66dc-47c4-8250-a9f25207c485)

Reply:
Much better, thanks @BominRahmani!